moveit_core/kinematics_base/include/moveit/kinematics_base/kinematics_base.h

```diff
@@ -388,6 +391,8 @@ class KinematicsBase
    * @param tip_frame The tip of the chain
    * @param search_discretization The discretization of the search when the solver steps through the redundancy
    * @return True if initialization was successful, false otherwise
+   *
+   * Note: this is intended to be removed in future ROS versions in favor of robot_model being passed in
```

Comment:
@baalexander suggested marking the below one as deprecated now, and then later removing it and marking this one as deprecated. That would take longer, but would have the advantage of not spewing warnings.